File: Harvard_CS50_Intro/Week 1/pset1/mario.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>

int main(void) {
    /* Getting input from user */
    int height;
    do {
        printf("Height: ");
        height = get_int();
    } while (height < 0 || height > 23);
    
    if (height == 0) {
        exit(0);
    }
    
    /* height = 4 */ 
    // index = 1 -> height;
    // print 
    // height - index " " and then index # then 2 " " then index # then height - index " "
    // height - (index + 1) " " and then index # then 2 " " then index # then height - (index + 1) " ";
    // .
    // .
    // .
    /* The total number of rows we need */
    for (int i = 1; i <= height; i++) {
        /* left-side space */
        for (int j = 0; j < height - i; j++) {
            printf(" ");
        }
        /* left-side hashes */
        for (int j = 0; j < i; j++) {
            printf("#");
        }
        /* middle space*/
        for (int j = 0; j < 2; j++) {
            printf(" ");
        }
        /* right-side space */
        for (int j = 0; j < i; j++) {
            printf("#");
        }
        /* This is too harsh, I guess for saving some space? :) */
        // for (int j = 0; j < height - i; j++) {
        //     printf(" ");
        // }
        printf("\n");
    }
}

File: Harvard_CS50_Intro/Week 8/pset6/mario.py
import cs50
import sys

def main():
    # Get the input Height
    # height = get_height()
    # print(height)

    # Print stairs
    print_stairs(get_height())

def get_height():
    while True:
        print("Height: ", end="")
        height = cs50.get_int()
        if height >= 1 and height <= 23:
            break
    return height   

def print_stairs(height):
    for i in range(1, height + 1):
        for j in range(height - i):
            print(" ", end="") # left-side space
        for j in range(i):
            print("#", end="") # left-side hashes
        for j in range(2):
            print(" ", end="") # middle space
        for j in range(i):
            print("#", end="") # right-side hashes
        print("")

if __name__ == "__main__":
    main()